app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -50,14 +51,20 @@ fun ConversationScreen(conversationViewModel: ConversationViewModel) {
         onMessageChanged = conversationViewModel::onMessageChanged,
         onSendButtonClicked = conversationViewModel::sendMessage,
         onSendAttachment = conversationViewModel::sendAttachmentMessage,
-        onDownloadAsset = conversationViewModel::downloadAsset,
-        onImageFullScreenMode = { conversationViewModel.navigateToGallery(it) },
+        onDownloadAsset = conversationViewModel::downloadOrFetchAsset,
+        onImageFullScreenMode = conversationViewModel::navigateToGallery,
         onBackButtonClick = conversationViewModel::navigateBack,
         onDeleteMessage = conversationViewModel::showDeleteMessageDialog,
         onCallStart = audioPermissionCheck::launch,
-        onError = conversationViewModel::onError
+        onSnackbarMessage = conversationViewModel::onSnackbarMessage
     )
     DeleteMessageDialog(conversationViewModel = conversationViewModel)
+    DownloadedAssetDialog(
+        conversationViewState = conversationViewModel.conversationViewState,
+        onSaveFileToExternalStorage = conversationViewModel::onSaveFile,
+        onOpenFileWithExternalApp = conversationViewModel::onOpenFileWithExternalApp,
+        hideOnAssetDownloadedDialog = conversationViewModel::hideOnAssetDownloadedDialog
```

Comment:
when we rename it to onDownloadCancel we could call a method inside the ViewModel like. : conversationViewModel::cancelDownloading() or something like that that way we are not stuck with a dialog implementation which indicates a function name but we can do whatever we want when user cancels downloading not only closing the dialog